erp/views.py

```diff
@@ -206,91 +214,102 @@ def _parse_location_or_404(lat, lon):
-    location = _parse_location_or_404(lat, lon)
-    qs = Erp.objects.select_related("accessibilite", "activite", "commune_ext").published().search_what(what)
-    commune = None
-    if commune_slug:
-        commune = get_object_or_404(Commune, slug=commune_slug)
-        qs = qs.filter(commune_ext=commune)
-        where = str(commune)
-    elif location:
-        qs = qs.nearest(location)
-    elif where and where not in ("France entière", translate("France entière")):
-        location = geocoder.autocomplete(where)
-        if location:
-            lat, lon = (location.y, location.x)  # update current searched lat/lon
-            qs = qs.nearest(location)
-        else:
-            qs = qs.search_commune(where)
-
-    return qs, {
-        "commune": commune,
-        "lat": lat,
-        "lon": lon,
-        "what": what,
-        "where": where,
-        "commune_json": commune.toTemplateJson() if commune else None,
+def _filter_erp_by_location(queryset, **kwargs):
+    search_type = kwargs.get("search_type")
+
+    if search_type == settings.IN_MUNICIPALITY_SEARCH_TYPE:
+        return queryset.filter(commune=kwargs.get("municipality").nom)
+
+    if search_type == settings.ADRESSE_DATA_GOUV_SEARCH_TYPE_CITY:
+        return queryset.annotate(city=Lower("commune")).filter(city=kwargs.get("municipality").lower())
+
+    if search_type in (settings.ADRESSE_DATA_GOUV_SEARCH_TYPE_HOUSENUMBER, "Autour de moi", translate("Autour de moi")):
+        lat, lon = kwargs.get("lat"), kwargs.get("lon")
+        location = _parse_location_or_404(lat, lon)
+        return queryset.nearest(location, max_radius_km=0.2)
+
+    if search_type == settings.ADRESSE_DATA_GOUV_SEARCH_TYPE_STREET:
+        street_name = kwargs.get("street_name")
+        return queryset.filter(code_postal=kwargs.get("postcode"), voie__icontains=street_name)
+    return queryset
+
+
+def search(request):
+    filters = _cleaned_search_params_as_dict(request.GET)
+    base_queryset = Erp.objects.published().search_what(filters.get("what"))
+    queryset = _filter_erp_by_location(base_queryset, **filters)
+
+    paginator = Paginator(queryset, 10)
+    pager = paginator.get_page(request.GET.get("page", 1))
+    pager_base_url = url.encode_qs(**filters)
+
+    context = {
+        **filters,
+        "pager": pager,
+        "pager_base_url": pager_base_url,
+        "geojson_list": make_geojson(pager),
+        "paginator": paginator,
     }
+    return render(request, "search/results.html", context=context)
 
 
-def search(request, commune_slug=None):
-    where, what, lat, lon, code = _clean_search_params(request.GET, "where", "what", "lat", "lon", "code")
-    where = where or "France entière"
-    qs, context = _get_search_qs_and_context(commune_slug, lat, lon, what, where)
-    # pager
-    paginator = Paginator(qs, 10)
+def search_in_municipality(request, commune_slug):
+    municipality = get_object_or_404(Commune, slug=commune_slug)
```

Comment:
Not sure we should based our search on this model. This supposed to have an up-to-date Commune database, which is not the case atm.

Reply:
I mostly rewrote what was already done I believe for this view. My proposition would be to analyze the issue further and create a ticket to check if we should:
- Try to remove this model
- Try to update the data frequently
- Leave everything as is